test-suite/shared/src/test/scala/org/scalajs/testsuite/javalib/lang/CharacterTest.scala

```diff
@@ -383,6 +383,432 @@ class CharacterTest {
+    assertEquals('0', Character.toUpperCase('0'))
+    assertEquals('A', Character.toUpperCase('A'))
+    assertEquals('A', Character.toUpperCase('a'))
+    assertEquals('F', Character.toUpperCase('f'))
+    assertEquals('Z', Character.toUpperCase('z'))
+    // compat/ucd
+    assertEquals('Β', Character.toUpperCase('β'))
+    assertEquals('Μ', Character.toUpperCase('μ'))
+    // alt (what is this?)
+    assertEquals('Ā', Character.toUpperCase('Ā'))
+    assertEquals('Ā', Character.toUpperCase('ā'))
+    assertEquals('Ă', Character.toUpperCase('ă'))
+    assertEquals('Į', Character.toUpperCase('į'))
+    // high
+    assertEquals('Ａ', Character.toUpperCase('ａ'))
+    assertEquals('Ｚ', Character.toUpperCase('ｚ'))
+    assertEquals('［', Character.toUpperCase('［'))
+  }
+
+  @Test def toUpperCase_Int(): Unit = {
+    // Low
+    assertEquals(10, Character.toLowerCase(10)) // '\n' => '\n'
+    // Ascii
+    assertEquals(65, Character.toUpperCase(65)) // A => A
+    assertEquals(65, Character.toUpperCase(97)) // a => A
+    assertEquals(70, Character.toUpperCase(102)) // f => F
+    assertEquals(90, Character.toUpperCase(122)) // z => Z
+    // compat/ucd
+    assertEquals(0x0392, Character.toUpperCase(0x03D0)) // β => Β
+    assertEquals(0x039C, Character.toUpperCase(0x00B5)) // μ => Μ
+    // alt
+    assertEquals(256, Character.toUpperCase(256))
+    assertEquals(256, Character.toUpperCase(257))
+    assertEquals(258, Character.toUpperCase(259))
+    assertEquals(302, Character.toUpperCase(303))
+    // high
+    assertEquals(65313, Character.toUpperCase(65345))
+    assertEquals(65338, Character.toUpperCase(65370))
+    assertEquals(65339, Character.toUpperCase(65339))
+  }
+
+/*
+def format(codePoint: Int): String = "0x%04x".format(codePoint)
+
+def myToUpperCase(cp: Int): Int = {
+  val upperString = new String(Array(cp),0,1).toUpperCase()
+  if (upperString.length() == 1)
+    upperString.charAt(0)
+  else
+    cp
+}
+
+for (cp <- 0 to Character.MAX_CODE_POINT) {
+  if (myToUpperCase(cp) != Character.toUpperCase(cp)) {
+    val cpStr: String  = new String(Array(cp),0,1)
+    val upperCP: Int = Character.toUpperCase(cp)
+    val upperStr: String = new String(Array(upperCP),0,1)
+
+    println(s"    assertEquals(${format(upperCP)}, Character.toUpperCase(${format(cp)})) // $cpStr => $upperStr")
+  }
+}
+*/
+  @Test def toUpperCase_Int_special(): Unit = {
+    assertEquals(0x1f88, Character.toUpperCase(0x1f80)) // ᾀ => ᾈ
+    assertEquals(0x1f89, Character.toUpperCase(0x1f81)) // ᾁ => ᾉ
+    assertEquals(0x1f8a, Character.toUpperCase(0x1f82)) // ᾂ => ᾊ
+    assertEquals(0x1f8b, Character.toUpperCase(0x1f83)) // ᾃ => ᾋ
+    assertEquals(0x1f8c, Character.toUpperCase(0x1f84)) // ᾄ => ᾌ
+    assertEquals(0x1f8d, Character.toUpperCase(0x1f85)) // ᾅ => ᾍ
+    assertEquals(0x1f8e, Character.toUpperCase(0x1f86)) // ᾆ => ᾎ
+    assertEquals(0x1f8f, Character.toUpperCase(0x1f87)) // ᾇ => ᾏ
+    assertEquals(0x1f98, Character.toUpperCase(0x1f90)) // ᾐ => ᾘ
```

Comment:
We should also test `0x1f88..0x1f8f` as *inputs* here, since our algorithm captures those as special case inputs. We need to make sure that our trick with the `| 0x0008` does the right thing for the inputs in that range that are already uppercase.

(same with the other 2 ranges of uppercases)

Reply:
Added this static ranger as `toUpperCase_CodePoint_special_cases`